go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -114,10 +113,22 @@ func tryCreateTxThrottler(config *tabletenv.TabletConfig, topoServer *topo.Serve
 		return nil, err
 	}
 
-	// Clone tsv.TxThrottlerHealthCheckCells so that we don't assume tsv.TxThrottlerHealthCheckCells
-	// is immutable.
-	healthCheckCells := make([]string, len(config.TxThrottlerHealthCheckCells))
-	copy(healthCheckCells, config.TxThrottlerHealthCheckCells)
+	var healthCheckCells []string
+	if len(config.TxThrottlerHealthCheckCells) > 0 {
+		// Clone tsv.TxThrottlerHealthCheckCells so that we don't assume tsv.TxThrottlerHealthCheckCells
+		// is immutable.
+		healthCheckCells = make([]string, len(config.TxThrottlerHealthCheckCells))
+		copy(healthCheckCells, config.TxThrottlerHealthCheckCells)
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), topo.RemoteOperationTimeout)
+		defer cancel()
+
+		var err error
+		healthCheckCells, err = topoServer.GetKnownCells(ctx)
```

Comment:
I'm curious if there are situations where `.GetKnownCells` might error or return zero cells when we don't want to, specifically on a cluster that is just coming online 🤔

Reply:
Another thing is if a new cell gets added, should transaction throttling starts monitoring those cells?

Reply:
@harshit-gangal great question, I wondered this too and I'm on the fence. I suppose the options are:
1. Document that a `vttablet` restart is needed if cells change
2. Make this code dynamic, depending on effort

I think it would be good to follow the themes of the rest of Vitess for consistency. Using `vtgate` as an example, does it begin to "watch" tablets in new cells if `--cells_to_watch` is unspecified 🤔?

For Slack we could probably accept that a rolling restart is needed, but avoiding it would be nice as well

Reply:
cells_to_watch acts as a filter, vtgate watches on all the cells with newly added ones.
The same theme should be followed for vttablet.

Reply:
> The same theme should be followed for vttablet.

@harshit-gangal agreed, I'm hoping to copy the same behaviour here and will give it a shot 👍 

I think the most efficient option would be to periodically check that we know all cells and update `healthCheckCell`, I think the only drawback is it will probably need a mutex if the checking is done async

Reply:
Agreed, this should be dynamic. @timvaillancourt this is also similar in behavior to `freno`, which periodically re-scanned the relevant targets. Also, see the tablet throttler implementation: 

https://github.com/vitessio/vitess/blob/11fd98da90c41bbfe522ec6761d731666978ab78/go/vt/vttablet/tabletserver/throttle/throttler.go#L777